File: intentfilterjava/src/main/java/com/example/intentfilterjava/MainActivity.java
package com.example.intentfilterjava;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import java.util.Locale;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }
    public void workWithIntent(View view){
        call();
    }
    public void sample(){
        Intent intent=new Intent();
        intent.setAction("com.example.action.COMMON");
        intent.setData(Uri.parse("tel:123456"));
        startActivity(intent);
    }
    public void dial(){
        Intent intent=new Intent(Intent.ACTION_DIAL);
        startActivity(intent);
    }
    public void call(){
        Intent intent=new Intent(Intent.ACTION_CALL);
        intent.setData(Uri.parse("tel://9175110108"));
        startActivity(intent);
    }
    public void web(){
        Intent intent=new Intent(Intent.ACTION_VIEW);
        intent.setData(Uri.parse("http://www.google.com"));
        startActivity(intent);
    }
    public void route(){
        String uri=String.format(Locale.ENGLISH,"https://maps.google.com");
        Intent intent=new Intent(Intent.ACTION_VIEW,Uri.parse(uri));
        intent.setPackage("com.google.android.app.maps");
        startActivity(intent);
    }

}

File: app/src/main/java/com/example/lenovo/intentfilterdemo/MainActivity.kt
package com.example.lenovo.intentfilterdemo

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.View

import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
        }
        btnOk.setOnClickListener{
            web()
        }
    }
    fun fun1(){
        val intent=Intent()
        intent.action="com.example.action.NEWS"
        intent.data= Uri.parse("http://www.google.com")
        startActivity(intent)
    }
    fun dial()=startActivity(Intent(Intent.ACTION_DIAL))

    fun call(){
        val intent=Intent(Intent.ACTION_CALL)
        intent.data= Uri.parse("tel://9503740614")
        startActivity(intent)
    }

    fun web(){
        val intent=Intent(Intent.ACTION_VIEW)
        intent.data= Uri.parse("http://www.google.com")
        startActivity(intent)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}
